src/Arcus.WebApi.Unit/Security/Authentication/SharedAccessKeyAuthenticationAttributeTests.cs

```diff
@@ -152,6 +215,15 @@ private async Task<HttpResponseMessage> SendAuthorizedHttpRequestWithHeader(stri
                 return await client.SendAsync(request);
             }
         }
+        private async Task<HttpResponseMessage> SendAuthorizedHttpRequestWithQuerystring(string parameterName, string parameterValue)
```

Comment:
Contains some duplication from `SendAuthorizedHttpRequestWithHeader`, we should try to refactor to reduce it by creating `SendAuthorizedHttpRequest` (or so) that creates the HttpClient and sends the provided HttpRequestMessages.

That would allow `SendAuthorizedHttpRequestWithQuerystring` to create an HttpRequestMessage with query parameter and `SendAuthorizedHttpRequestWithHeader` to add the HTTP headers

Reply:
Or just provide capability to add headers and query parameters to one method that handles everything.

This will also be required for a new test that I'd like to add

Reply:
will look at this

Reply:
@tomkerkhove This has been updated in latest PR

Reply:
Looks good, thanks!